File: solutions/zig/eliuds-eggs/1/eliuds_eggs.zig
pub fn eggCount(number: usize) u4 {
    var count: u4 = 0;
    var n = number;

    while (n > 0) {
        if (n & 1 == 1) {
            count += 1;
        }
        n >>= 1;
    }

    return count;
}

File: solutions/typescript/eliuds-eggs/1/eliuds-eggs.ts
export const eggCount = (number: number): number => {
    let count = 0;
    let n = number;
    while (n > 0) {
        if ((n & 1) === 1) {
            count += 1;
        }
        n >>= 1;
    }
    return count;
};
